modules/head-node/templates/fleet_config.yaml.tpl

```diff
@@ -29,7 +29,7 @@ jenkins:
         oldId: "803939ae-c6f8-4c7a-8386-0d1a9af631ad"
         privateIpUsed: false
         region: "us-west-2"
-        restrictUsage: false
+        restrictUsage: true
```

Comment:
I think we handle this within the Jenkinsfile when nodes are allocated (e.g. https://github.com/apache/tvm/blob/main/Jenkinsfile#L173 should only go to a node with the `CPU` label), but from the help description of this it sounds like it does the same thing

Reply:
This would enforce that a label expression must match for the node to be used. For example, if a new stage was added that didn't have any label expression it wouldn't be able to schedule onto a node until it was properly labeled.